setup.py

```diff
@@ -774,24 +771,28 @@ def write_version(filename: str = os.path.join(*[my_dir, "airflow", "git_version
     'zendesk': ["zendesk"],
 }
 
-
-# Make devel_all contain all providers + extras + unique
-devel_all = list(
+# All "users" extras (no devel extras)
+all_ = list(
     set(
-        devel
-        + [req for req_list in EXTRAS_REQUIREMENTS.values() for req in req_list]
+        [req for req_list in EXTRAS_REQUIREMENTS.values() for req in req_list]
         + [req for req_list in PROVIDERS_REQUIREMENTS.values() for req in req_list]
     )
 )
+EXTRAS_REQUIREMENTS.update({"all": all_})
 
-PACKAGES_EXCLUDED_FOR_ALL = []
+# Development extras
+EXTRAS_REQUIREMENTS.update(
+    {
+        'devel': devel_minreq,  # includes doc
+        'devel_hadoop': devel_hadoop,  # includes devel_minreq
+        'doc': doc,
+    }
+)
+# This can be simplify to devel_hadoop + all_ due to inclusions
+# but we keep it for explicit sake
+devel_all = list(set(all_ + doc + devel_minreq + devel_hadoop))
 
-if PY3:
-    PACKAGES_EXCLUDED_FOR_ALL.extend(
-        [
-            'snakebite',
-        ]
-    )
```

Comment:
For 2.0 we are using only Py3+ so the if is always true. Moreover there's no `snakebite` requirement in any extra.